server/src/queries/mod.rs

```diff
@@ -1,29 +1,78 @@
     models::{BlockStatus, DBResponse},
-    router::QueryRequestFilter,
-    ApiContext,
+    router::{Build, QueryRequestFilter},
+    ApiContext, Config, SubCommand,
 };
-use axum::Extension;
-use sqlx::{Pool, Postgres};
-
-pub async fn get_latest_blockheight(
-    ctx: &Extension<ApiContext>,
-    network: &QueryRequestFilter,
-) -> Result<i64, sqlx::Error> {
-    let row: (i64,) = sqlx::query_as("SELECT MAX(height) FROM blocks;")
-        .fetch_one(match network {
-            QueryRequestFilter::Mainnet => &ctx.mainnet_db,
-            _ => &ctx.devnet_db,
+
+use anyhow::Context;
+use axum::{http::Method, Extension, Router};
+use mockall::predicate::*;
+use mockall::*;
+use sqlx::postgres::PgPoolOptions;
+use tower::ServiceBuilder;
+use tower_http::cors::{Any, CorsLayer};
+
+#[automock]
+impl ApiContext {
+    pub async fn new(config: Config) -> anyhow::Result<Self> {
+        let mainnet_ledger = Ledger::init_async(&config.mainnet_ledger_path)
+            .await
+            .expect("Error: Could not create mainnet ledger.");
+
+        let devnet_ledger = Ledger::init_async(&config.devnet_ledger_path)
+            .await
+            .expect("Error: Could not create devnet ledger.");
+
+        let mainnet_db = PgPoolOptions::new()
+            .max_connections(25)
+            .connect(&config.mainnet_database_url)
+            .await
+            .context("Error: Could not connect to mainnet database.")?;
+
+        let devnet_db = PgPoolOptions::new()
+            .max_connections(25)
+            .connect(&config.devnet_database_url)
+            .await
+            .context("Error: Could not connect to devnet database.")?;
+
+        Ok(ApiContext {
+            config: Arc::new(config),
+            mainnet_ledger: Arc::new(mainnet_ledger.db),
+            devnet_ledger: Arc::new(devnet_ledger.db),
+            mainnet_db,
+            devnet_db,
         })
-        .await?;
-    Ok(row.0)
-}
+    }
 
-/// By default we get all signals every query.
-/// This is wildly inefficient & we should work towards an alternative.
-/// Since we're essentially only interested in memo's - maybe a way to decode base58 on the DB side?
-pub async fn get_signals(db: &Pool<Postgres>) -> Result<Vec<DBResponse>, sqlx::Error> {
-    sqlx::query_as!(DBResponse,
+    pub async fn get_latest_block_height(
+        &self,
+        network: &QueryRequestFilter,
+    ) -> Result<i64, sqlx::Error> {
+        let row: (i64,) = sqlx::query_as("SELECT MAX(height) FROM blocks;")
+            .fetch_one(match network {
+                QueryRequestFilter::Mainnet => &self.mainnet_db,
+                _ => &self.devnet_db,
+            })
+            .await?;
+        Ok(row.0)
+    }
+
+    /// By default we get all signals every query.
+    /// This is wildly inefficient & we should work towards an alternative.
+    /// Since we're essentially only interested in memo's - maybe a way to decode base58 on the DB side?
+    pub async fn get_signals(
```

Comment:
This function should not be part of the APIContext - it's a query & not a direct method on the context itself.